gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_custom_metric.ipynb

```diff
@@ -0,0 +1,1390 @@
+        "\n",
+        "This notebook demostrates how to leverage Vertex AI Prompt Optimizer (Preview) to optimize a simple prompt for a Gemini model using your own metric. The goal is to use Vertex AI Prompt Optimizer (Preview) to find the new prompt template which generates responses according to your own metric.\n",
+        "\n",
+        "\n",
+        "This tutorial uses the following Google Cloud ML services and resources:\n",
+        "\n",
+        "- Vertex Gen AI\n",
+        "- Vertex AI Prompt Optimizer (Preview)\n",
+        "- Vertex AI Model Eval\n",
+        "- Vertex AI Custom job\n",
+        "- Cloud Run\n",
+        "\n",
+        "The steps performed include:\n",
+        "\n",
+        "- Prepare the prompt-ground truth pairs optimized for another model\n",
+        "- Define the prompt template you want to optimize\n",
+        "- Define and deploy your own custom evaluation metric on Cloud function\n",
+        "- Set optimization mode and steps\n",
+        "- Run the automatic prompt optimization job\n",
+        "- Collect the best prompt template and eval metric\n",
+        "- Validate the best prompt template"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "08d289fa873f"
+      },
+      "source": [
+        "### Dataset\n",
+        "\n",
+        "The dataset is a question-answering dataset generated by  a simple AI cooking assistant that provides suggestions on how to cook healthier dishes.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "aed92deeb4a0"
+      },
+      "source": [
+        "### Costs\n",
+        "\n",
+        "This tutorial uses billable components of Google Cloud:\n",
+        "\n",
+        "* Vertex AI\n",
+        "* Cloud Storage\n",
+        "\n",
+        "Learn about [Vertex AI pricing](https://cloud.google.com/vertex-ai/pricing) and [Cloud Storage pricing](https://cloud.google.com/storage/pricing) and use the [Pricing Calculator](https://cloud.google.com/products/calculator/) to generate a cost estimate based on your projected usage."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "61RBz8LLbxCR"
+      },
+      "source": [
+        "## II. Before you start"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "No17Cw5hgx12"
+      },
+      "source": [
+        "### Install Vertex AI SDK for Python and other required packages\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "tFy3H3aPgx12"
+      },
+      "outputs": [],
+      "source": [
+        "! pip3 install --upgrade --quiet 'google-cloud-aiplatform[evaluation]'\n",
+        "! pip3 install --upgrade --quiet 'plotly'\n",
+        "! pip3 install --upgrade --quiet 'asyncio' 'tqdm' 'tenacity' 'etils' 'importlib_resources' 'fsspec' 'gcsfs' 'nbformat>=4.2.0'"
+      ]
```

Comment:
Can you put these in one install command?

Reply:
done